File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/plugin/PluginArtifactRepository.java
package com.mieai.qqbot.persistence.plugin;

import java.util.List;
import java.util.Optional;

public interface PluginArtifactRepository {
    void upsert(PluginArtifact artifact);
    Optional<PluginArtifact> findById(String pluginId);
    List<PluginArtifact> findAll();
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/plugin/PluginArtifactRepository.kt
package com.mieai.qqbot.persistence.plugin

interface PluginArtifactRepository {
    fun upsert(artifact: PluginArtifact)

    fun findById(pluginId: String): PluginArtifact?

    fun findAll(): List<PluginArtifact>
}
